compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -3016,6 +3016,7 @@ fn bind_generator_hidden_types_above<'tcx>(
                         counter += 1;
                         tcx.mk_re_late_bound(current_depth, br)
                     }
+                    ty::RePlaceholder(_) => r,
```

Comment:
Hmm... I don't think we're supposed to leak placeholder types here. They should all be erased here...

Reply:
We perform substitution with solver-provided `substs`, so we can have any region that is allowed within `substs`.
It may be interesting to do the fold before substitution: substs cannot contain ReErased, and unsubstituted `bty` can only contain ReErased.

Reply:
oh, so this is a case where we had a late bound lifetime that was bound deeper than the generator itself. huh, ok, maybe this is fine then.